tool.go

```diff
@@ -784,18 +794,111 @@ type testFunc struct {
 	Output  string // output, for examples
 }
 
+var testFileSet = token.NewFileSet()
+
+func (t *testFuncs) load(filename, pkg string, doImport, seen *bool) error {
+	f, err := parser.ParseFile(testFileSet, filename, nil, parser.ParseComments)
+	if err != nil {
+		return err
+	}
+	for _, d := range f.Decls {
+		n, ok := d.(*ast.FuncDecl)
+		if !ok {
+			continue
+		}
+		if n.Recv != nil {
+			continue
+		}
+		name := n.Name.String()
+		switch {
+		case isTestMain(n):
+			if t.TestMain != nil {
+				return errors.New("multiple definitions of TestMain")
+			}
+			t.TestMain = &testFunc{pkg, name, ""}
+			*doImport, *seen = true, true
+		case isTest(name, "Test"):
+			t.Tests = append(t.Tests, testFunc{pkg, name, ""})
+			*doImport, *seen = true, true
+		case isTest(name, "Benchmark"):
+			t.Benchmarks = append(t.Benchmarks, testFunc{pkg, name, ""})
+			*doImport, *seen = true, true
+		}
+	}
+	// 	ex := doc.Examples(f)
+	// 	sort.Sort(byOrder(ex))
+	// 	for _, e := range ex {
+	// 		*doImport = true // import test file whether executed or not
+	// 		if e.Output == "" && !e.EmptyOutput {
+	// 			// Don't run examples with no output.
+	// 			continue
+	// 		}
+	// 		t.Examples = append(t.Examples, testFunc{pkg, "Example" + e.Name, e.Output})
+	// 		*seen = true
+	// 	}
```

Comment:
Is this the code to support examples that's commented out because it's blocked on #381, right?

I have a simple request. Either remove the code and just make note of it in the PR (in a comment or so), so we can look it up and add it later after #381 is resolved.

Either that, or, if you want to keep it, then please add a comment explaining why this is commented out. E.g.:

```Go
// This is the code to support examples. It's blocked on https://github.com/gopherjs/gopherjs/issues/381 being resolved.
/*ex := doc.Examples(f)
sort.Sort(byOrder(ex))
for _, e := range ex {
	*doImport = true // import test file whether executed or not
	if e.Output == "" && !e.EmptyOutput {
		// Don't run examples with no output.
		continue
	}
	t.Examples = append(t.Examples, testFunc{pkg, "Example" + e.Name, e.Output})
	*seen = true
}*/
```

If you're indifferent, I prefer removing it (just make sure a copy of it is accessible somewhere in this PR for lookup at a later time).